lib/folio/status_current_location.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Folio
+  # Folio::StatusCurrentLocation takes an item
+  # and any requests associated with the item's instance record
+  # and creates an equivalent Symphony current location
+  class StatusCurrentLocation
+    attr_reader :item, :requests
+
+    def initialize(item, requests)
+      @item = item
+      @requests = requests
+    end
+
+    def current_location
+      case status
+      when 'Checked out', 'Claimed returned', 'Aged to lost'
+        'CHECKEDOUT'
+      when 'Awaiting pickup'
+        symphony_pickup_location_code
+      when 'In process', 'In process (non-requestable)'
+        'INPROCESS'
+      when 'In transit', 'Awaiting delivery'
+        'INTRANSIT'
+      when 'Missing', 'Long missing'
+        'MISSING'
+      when 'On order'
+        'ON-ORDER'
+      end
+    end
+
+    private
+
+    def status
+      item['status']
+    end
+
+    # ARS-LOAN, RUM-LOAN, and SPE-LOAN each receives a
+    # special label in SearchWorks. Other codes are generically
```

Comment:
Where will these labels come from in Searchworks? Aren't we moving toward using Folio location labels?

Reply:
The equivalent in FOLIO for these locations would be service points, which I don't think we're doing anything with in SearchWorks yet. We might want to. But this gets us to parity with Symphony by relying on the existing `Constants:LOCS` mappings.

Reply:
@cbeer I'm confused about what the plan is for `Constants::LOCS` vs FOLIO display names. My guess is we're still going to have location codes that don't map to anything in FOLIO and we'll need to maintain a fallback list in SearchWorks (maybe I'm wrong about this). This PR is a refinement to previous work to map folio statuses to symphony locations so they'll display right in searchworks. Is this no longer what we want to do?

See:

- https://github.com/sul-dlss/searchworks_traject_indexer/pull/940/files
- https://github.com/sul-dlss/searchworks_traject_indexer/pull/973/files

Reply:
Tried to capture post-folio work related to this in #1026